### PR TITLE
Gradle Scan Link Metric 

### PR DESCRIPTION
**Pull request by cdsap**

### CODE REVIEW

File: library/core/talaiot/src/main/kotlin/com/cdsap/talaiot/metrics/GradleScanLinkMetric.kt [hunk 1]
```diff
@@ -24,39 +24,6 @@ class GradleScanLinkMetric : BuildResultMetric<String?>(
         val services = gradleInternal.services
 
         when {
-            GradleVersion.current() > GradleVersion.version("5.0")
```
Comment (Malinskiy) [inline library/core/talaiot/src/main/kotlin/com/cdsap/talaiot/metrics/GradleScanLinkMetric.kt:27]: So we don't provide a way to use the new plugin with the old versions of gradle? This sounds just like the approach a lot of plugins are choosing and they use the same version string for the plugin as gradle, e.g. gradle 6.7 requires plugin version 6.7

Comment (cdsap) [reply library/core/talaiot/src/main/kotlin/com/cdsap/talaiot/metrics/GradleScanLinkMetric.kt:27]: We don't provide support for Gradle < 6.  

Sounds wrong, but the current build system's state and the main components of the build stack(agp/kgp) force you to be updated and prepare for the next features. 
Talking about AS/AGP we have:
AGP 4.0.0+ -- Gradle 6.1.1+
AGP 4.1.0+ -- Gradle 6.5+
If one project wants to build with AS 4.1, use agp new features or build improvements, it's forced to update Gradle 6.5. 
Google is doing a great job creating hype with Compose and forcing the teams to talk about build possibilities to integrate compose, translated in embrace newer build tools.

That's why in the short-mid term, I'm focusing in be compatible with Gradle 7 and the Configuration Cache feature(Google is doing a great job creating hype with Compose and forcing the teams to talk about build possibilities to integrate compose, agp, kgp). 

I will add a couple of E2E tests and put ready for review

Comment (Malinskiy) [reply library/core/talaiot/src/main/kotlin/com/cdsap/talaiot/metrics/GradleScanLinkMetric.kt:27]: That branch condition also include [6.0, 6.7), I'm kinda interested in those versions as well as I expect they're still used

Comment (cdsap) [reply library/core/talaiot/src/main/kotlin/com/cdsap/talaiot/metrics/GradleScanLinkMetric.kt:27]: added back support but from 6.0.
Added e2e tests, ready for review

File: library/core/talaiot/build.gradle.kts [hunk 2]
```diff
@@ -14,7 +14,7 @@ dependencies {
     api(project(":library:core:talaiot-logger"))
     api(project(":library:core:talaiot-request"))
     implementation("com.github.oshi:oshi-core:5.5.0")
-    implementation("com.gradle:gradle-enterprise-gradle-plugin:3.5.2")
+    api("com.gradle:gradle-enterprise-gradle-plugin:3.5.2")
```
Comment (Malinskiy) [inline library/core/talaiot/build.gradle.kts:17]: Probably worth mentioning that we depend on this version. If the use will upgrade to later one - there are no contracts defined and our implementation might not work

**Review by Malinskiy — APPROVED** [review]

File: library/core/talaiot/build.gradle.kts (hunk 2 above)
Comment (cdsap) [reply library/core/talaiot/build.gradle.kts]: good catch, thanks. Reverted

**Comment by cdsap** [comment]
@Malinskiy the base branch is your branch, can I change back to master?

**Comment by Malinskiy** [comment]
Sure thing @cdsap, makes more sense to target master

**Pull request merged**